mule-extensions-api/src/main/java/org/mule/runtime/extension/api/dsl/syntax/DslSyntaxUtils.java

```diff
@@ -94,8 +95,11 @@ static boolean isFlattened(ObjectFieldType field, MetadataType fieldValue) {
     return fieldValue instanceof ObjectType && field.getAnnotation(FlattenedTypeAnnotation.class).isPresent();
   }
 
-  static String getTypeKey(MetadataType type, String namespace, String namespaceUri) {
-    return getId(type).get() + namespace + namespaceUri;
+  static Optional<String> getTypeKey(MetadataType type, String namespace, String namespaceUri) {
+    if (getId(type).isPresent()) {
+      return Optional.of(getId(type).get() + namespace + namespaceUri);
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Call "Optional#isPresent()" before accessing the value. ([squid:S3655](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS3655))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule-extensions-api&pullRequest=629&issues=AWiWQrPmotIXnmsU-R3L&open=AWiWQrPmotIXnmsU-R3L)

[ISSUE_KEY]: AWiWQrPmotIXnmsU-R3L